### DISCUSSION TITLE
Custom Script ORM vs REST API

### DISCUSSION THREAD

**Discussion by wz4** [netbox-community/netbox#14737] [Q&A] [answered]
When writing custom scripts, is it considered acceptable to still use the REST API to interact with Netbox objects, or is it best practice to always use ORM?

**Comment by candlerb** [comment, marked as answer]
I'd say best practice is to use ORM:
* You get transactional safety and rollback (note that "dry run" functionality depends on this)
* You don't need to hard-code an API key into the script

You also have a wider range of Django ORM filtering facilities available.
